File: backend/src/main/java/com/techbloghub/entity/Blog.java
package com.techbloghub.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "blogs")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Blog {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, unique = true)
    private String name;
    
    @Column(nullable = false)
    private String company;
    
    @Column(nullable = false, unique = true)
    private String rssUrl;
    
    @Column(nullable = false)
    private String siteUrl;
    
    @Column
    private String description;
    
    @Column
    private String logoUrl;
    
    @Enumerated(EnumType.STRING)
    @Builder.Default
    private BlogStatus status = BlogStatus.ACTIVE;
    
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    @Column(name = "last_crawled_at")
    private LocalDateTime lastCrawledAt;
    
    @OneToMany(mappedBy = "blog", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Post> posts;
    
    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
    
    public enum BlogStatus {
        ACTIVE, INACTIVE, PENDING_APPROVAL
    }
}

File: python-crawler/app/models/blog.py
from sqlalchemy import Column, Integer, String, DateTime, Enum as SQLEnum, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from enum import Enum
from ..core.database import Base


class BlogStatus(str, Enum):
    ACTIVE = "ACTIVE"
    INACTIVE = "INACTIVE" 
    SUSPENDED = "SUSPENDED"


class BlogType(str, Enum):
    WORDPRESS = "WORDPRESS"
    NAVER_D2 = "NAVER_D2" 
    NHN_CLOUD = "NHN_CLOUD"
    LYCORP = "LYCORP"
    MEDIUM = "MEDIUM"
    KAKAO = "KAKAO"
    TOSS = "TOSS"


class BlogEntity(Base):
    __tablename__ = "blog"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False, unique=True, index=True)
    company = Column(String, nullable=False)
    rss_url = Column(String, nullable=False, unique=True)
    site_url = Column(String, nullable=False)
    description = Column(Text)
    logo_url = Column(String)
    status = Column(SQLEnum(BlogStatus), default=BlogStatus.ACTIVE)
    blog_type = Column(SQLEnum(BlogType), default=None)
    last_crawled_at = Column(DateTime)
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), default=func.now(), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), default=func.now(), onupdate=func.now())
    
    # Relationships
    posts = relationship("PostEntity", back_populates="blog", cascade="all, delete-orphan")